data/lib/service/match/match_service.dart

```diff
@@ -568,19 +577,29 @@ class MatchService {
           id: user.id,
         );
       }).toList();
-
-      return squad;
     } catch (error, stack) {
       throw AppError.fromError(error, stack);
     }
   }
 
   Future<List<MatchModel>> getMatchesByIds(List<String> matchIds) async {
     try {
-      return await _matchCollection
+      if (matchIds.length > 10) {
+        throw AppError.fromError(
```

Comment:
this'll throw error and then show error screen or toast, right? This can be handled by us efficiently instead of showing user some error.

You can refer to `getUsersByIds` in UserServices where we chunked list and then fetched data as we needed.